### PR TITLE
NServiceBus.Metrics.ServiceControl

### PR DESCRIPTION
**Pull request by seanfarmar**
related to https://github.com/Particular/NServiceBus.Metrics.ServiceControl/issues/2

### CODE REVIEW

**Comment by Scooletz** [comment]
Lovely, we've got a 💚 build. Now it's time to create the solution and the page content

**Comment by seanfarmar** [comment]
@SzymonPobiega please check this out?

@Particular/docs-maintainers is the test failing because of something I pushed?

File: nservicebus/operations/metrics-service-control.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+summary: Reporting metrics to ServiceControl.
+reviewed: 2017-09-26
+component: MetricsServiceControl
+related:
+ - nservicebus/operations
+related:
+ - samples/logging/metrics
+---
+
+## WIP
```
Comment (Scooletz) [inline nservicebus/operations/metrics-service-control.md]: Could you write the content of this page? I know that we wait for the release decision, but once the content is done and a package released we can move forward with it.

**Comment by DavidBoike** [comment]
@seanfarmar It's from a content verification that was enabled after you branched. A rebase will fix.

File: nservicebus/operations/metrics-service-control.md (hunk 1 above)
Comment (seanfarmar) [reply nservicebus/operations/metrics-service-control.md]: I had a session with @tmasternak (Thanks Tomasz!) and pushed an updated version now

File: nservicebus/operations/metrics-service-control.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+The default is "particular.monitoring" which is ServiceControl monitoring instance input queue.
+
+
+#### Instance Id:
+
+An override for `$.diagnostics.hostid` and `$.diagnostics.hostdisplayname`.
+
+By default the monitoring plug-in will use [host identifiers](/nservicebus/hosting/override-hostid.md) to identify the monitored endpoint instance.
+
+The value of `$.diagnostics.hostdisplayname` is expected to be human readable. This might be problematic when using the [Azure Host](/nservicebus/hosting/cloud-services-host/faq.md#host-identifier) as it uses Role Name for the `$.diagnostics.hostdisplayname`, which results in the same display name for all instances.
```
Comment (SimonCropp) [inline nservicebus/operations/metrics-service-control.md]: given the cloud-services-host is becoming obsolete i dont think that is a good place to send people for more information on azure hosting

Comment (tmasternak) [reply nservicebus/operations/metrics-service-control.md]: I agree it's not perfect. But `instanceId` exists pretty much only because of this :/.

File: nservicebus/operations/metrics-service-control.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+## Enabling
+
+It can be enabled via:
+
+snippet: SendMetricDataToServiceControl
+
+
+#### Service Control Metrics Address:
```
Comment (tmasternak) [inline nservicebus/operations/metrics-service-control.md]: Do we need semicolons here and at the end of next section?

File: nservicebus/operations/metrics-service-control.md (hunk 2 above)
Comment (SimonCropp) [reply nservicebus/operations/metrics-service-control.md]: but people can host in a cloud-services without using cloud-services-host. so you are saying in that case they do not need the instanceid?

Comment (tmasternak) [reply nservicebus/operations/metrics-service-control.md]: Yes. The only scenario that required this argument I know of is CloudService host.

File: nservicebus/operations/metrics-service-control.md (hunk 3 above)
Comment (seanfarmar) [reply nservicebus/operations/metrics-service-control.md]: I'll remove them

File: nservicebus/operations/metrics-service-control.md (hunk 2 above)
Comment (seanfarmar) [reply nservicebus/operations/metrics-service-control.md]: There is another use for that, for deployment reasons as described here: https://github.com/Particular/ServiceControl/issues/329

**Comment by seanfarmar** [comment]
@Scooletz can you re-review please?

File: nservicebus/operations/metrics-service-control.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+reviewed: 2017-09-26
+component: MetricsServiceControl
+related:
+ - nservicebus/operations
+related:
+ - samples/logging/metrics
+---
+
+
+This component enables sending monitoring data to ServiceControl.Monitoring instance.
```
Comment (SimonCropp) [inline nservicebus/operations/metrics-service-control.md]: in this context "ServiceControl.Monitoring" is ambiguous and should link to more info

File: nservicebus/operations/metrics-service-control.md (hunk 2 above)
Comment (SeanFeldman) [reply nservicebus/operations/metrics-service-control.md]: WebJobs would have a similar problem. Scale happens automatically and instance ids are not deterministic. [Link to GG](https://groups.google.com/forum/#!searchin/particularsoftware/webjob$20host$20id%7Csort:relevance/particularsoftware/URnuGDf27QM/TCPsGxD9AAAJ)

Comment (tmasternak) [reply nservicebus/operations/metrics-service-control.md]: @SeanFeldman just to validate my assumptions. I though that Cloud Service host was the only problematic piece as it was doing a custom HostId, HostDisplayName override at the header level. 

My understanding was that even if I host using `WebJobs` I could use [host override api](https://docs.particular.net/nservicebus/hosting/override-hostid?version=core_7#overriding-the-host-identifier) and set:
 * `HostId` to deterministic GUID
 * `HostDisplayName` to some human readable thing based on rolename

The problem with CloudService host as I understood it was that it would override whatever I would set using custom hostid api.

File: Snippets/MetricsServiceControl/MetricsServiceControl_1/Configuration.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    using NServiceBus.Metrics.ServiceControl;
+
+    class Configuration
+    {
+        void ConfigureEndpoint(BusConfiguration busConfiguration)
+        {
+            #region SendMetricDataToServiceControl
+
+            busConfiguration.SendMetricDataToServiceControl(
+                serviceControlMetricsAddress: "address",
```
Comment (Scooletz) [inline Snippets/MetricsServiceControl/MetricsServiceControl_1/Configuration.cs]: Should we use the default queue value like in https://github.com/Particular/ServiceControl.Monitoring/pull/62

File: Snippets/MetricsServiceControl/MetricsServiceControl_2/Configuration.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    using NServiceBus.Metrics.ServiceControl;
+
+    class Configuration
+    {
+        void ConfigureEndpoint(EndpointConfiguration endpointConfiguration)
+        {
+            #region SendMetricDataToServiceControl
+
+            endpointConfiguration.SendMetricDataToServiceControl(
+                serviceControlMetricsAddress: "address",
```
Comment (Scooletz) [inline Snippets/MetricsServiceControl/MetricsServiceControl_2/Configuration.cs]: Should we use the default queue value like in https://github.com/Particular/ServiceControl.Monitoring/pull/62

File: Snippets/MetricsServiceControl/Metrics_1/Configuration.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    class Configuration
+    {
+        void ConfigureEndpoint(EndpointConfiguration endpointConfiguration)
+        {
+            #region SendMetricDataToServiceControl
+
+            var metrics = endpointConfiguration.EnableMetrics();
+#pragma warning disable 618
+            metrics.SendMetricDataToServiceControl(
+                serviceControlMetricsAddress: "address",
```
Comment (Scooletz) [inline Snippets/MetricsServiceControl/Metrics_1/Configuration.cs]: Should we use the default queue value like in https://github.com/Particular/ServiceControl.Monitoring/pull/62

File: nservicebus/operations/metrics-service-control.md (hunk 4 above)
Comment (Scooletz) [reply nservicebus/operations/metrics-service-control.md]: > This component enables sending monitoring data to ServiceControl.Monitoring instance.

I'd rephrase it to:

> This component enables sending monitoring data gathered with NServiceBus.Metrics to an instance of ServiceControl.Monitoring service.

File: nservicebus/operations/metrics-service-control.md (hunk 2 above)
Comment (SeanFeldman) [reply nservicebus/operations/metrics-service-control.md]: Not entirely sure. @yvesgoeleven do you recall if that's what CS host did?

**Comment by Scooletz** [comment]
I'm ok with the changes. Just saw the comment related to the `instanceId`. Do we need to address it now or can it be a separate PR @seanfarmar ?

**Comment by seanfarmar** [comment]
@Scooletz I can updated it now if you think it won't stop this from moving forward?

**Comment by weralabaj** [comment]
@Scooletz @seanfarmar is that ready to be reviewed by maintainers & merged or still a WIP? :)

**Comment by Scooletz** [comment]
I think it's not ready yet. Yesterday we were red on all builds of these packages. 
Let me go through the branches and versions one more time. I'll update this issue as soon as I'm done with checks.

**Comment by Scooletz** [comment]
@seanfarmar Can you bump up the dependency for v5 to stable `1.0.0` and verify that the nuget dependency is right now?

**Comment by Scooletz** [comment]
@seanfarmar We want NSB.Metrics.SC 3.0 in here, right? https://github.com/Particular/NServiceBus.Metrics.ServiceControl/pull/5

**Comment by weralabaj** [comment]
@Scooletz really confusing :D If you're still changing stuff can you mark it as a WIP?

**Comment by Scooletz** [comment]
I did not unWIP it @weralabaj Changed again to WIP

**Comment by Scooletz** [comment]
@seanfarmar If you hit any obstacles with it, just ping me.

**Comment by seanfarmar** [comment]
@weralabaj this is now ready for review please :-)

File: nservicebus/operations/metrics-service-control.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: Seems that new page is not included in the menu, is that intentional?
If not, please add a new entry in menu.yml.

File: nservicebus/operations/metrics-service-control.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+title: Metrics data to ServiceControl
+summary: Reporting metrics to ServiceControl.
+reviewed: 2017-09-26
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: We can bump the date to today's.

File: nservicebus/operations/metrics-service-control.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+component: MetricsServiceControl
+related:
+ - nservicebus/operations
+related:
+ - samples/logging/metrics
+---
+
+
+This component enables sending monitoring data gathered with NServiceBus.Metrics to an instance of ServiceControl.Monitoring service.
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: Package names should be inside ticks (i.e. `)

File: nservicebus/operations/metrics-service-control.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+## Enabling
+
+It can be enabled via:
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: I'd change the title from "Enabling" to "Configuration" and then:

1. Say whether it's enabled by default or not, e.g. "After installation/adding package to the solution, metrics are/won't be automatically sent to SC" or something like this.
2. Keep what you have here, i.e. "It can be enabled via: + snippet"

File: nservicebus/operations/metrics-service-control.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+snippet: SendMetricDataToServiceControl
+
+
+#### Service Control Metrics Address
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: I think that should be a 3-rd level header, so #### => ###

File: nservicebus/operations/metrics-service-control.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+The default is "particular.monitoring" which is ServiceControl monitoring instance input queue.
+
+
+#### Instance Id
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: I think that should be a 3-rd level header, so #### => ###

File: nservicebus/operations/metrics-service-control.md [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+## Enabling
+
+It can be enabled via:
+
+snippet: SendMetricDataToServiceControl
+
+
+#### Service Control Metrics Address
+
+The default is "particular.monitoring" which is ServiceControl monitoring instance input queue.
```
Comment (weralabaj) [inline nservicebus/operations/metrics-service-control.md]: Here also we can use ticks instead of quotes, I think it'll look better in terms of formatting so " => `

File: nservicebus/operations/metrics-service-control.md (hunk 2 above)
Comment (weralabaj) [reply nservicebus/operations/metrics-service-control.md]: Seems the original concern here hasn't been addressed, is that intentional or fell through the cracks?

**Comment by weralabaj** [comment]
@seanfarmar I had some minor comments, let me know if anything is not clear.

**Comment by seanfarmar** [comment]
@weralabaj can you explain 
>Seems the original concern here hasn't been addressed, is that intentional or fell through the cracks?
please

File: nservicebus/operations/metrics-service-control.md (hunk 2 above)
Comment (seanfarmar) [reply nservicebus/operations/metrics-service-control.md]: We have dropped any reference to `NSerfviceBus.Hosthost` and only included the relevant Azure Host related info
